explos/channel-manager/sqlLite.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"fmt"
+	sqlite "github.com/flyingtime/gorm-sqlcipher"
+	"gorm.io/gorm"
+)
+
+type sqlLite struct {
+	db *gorm.DB
+}
+
+func NewSqliteDB() (*sqlLite, error) {
+	db, err := gorm.Open(sqlite.Open("test.db"))
+	if err != nil {
+		return nil, err
+	}
+
+	s := &sqlLite{db: db}
+	err = s.db.AutoMigrate(&User{}, &TeritoriKey{}, &Workspace{}, &Channel{})
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+type Workspace struct {
+	ID       uint `gorm:"primary_key"`
+	Name     string
+	Channels []Channel `gorm:"ForeignKey:Wid"`
+}
+
+type Channel struct {
+	ID          uint `gorm:"primary_key"`
+	BertyLink   string
+	ChannelName string
+	Wid         uint
+}
+
+type TeritoriKey struct {
+	ID     uint `gorm:"primary_key"`
+	PubKey string
+	UserId uint
+}
+
+type User struct {
+	ID              uint `gorm:"primary_key"`
+	BertyPubKey     string
+	TerritoriPubKey []TeritoriKey `gorm:"ForeignKey:UserId"`
+}
+
+func (s sqlLite) AddUser(bertyPubKey string) error {
+	db := s.db
+	tx := db.FirstOrCreate(&User{
+		BertyPubKey: bertyPubKey,
+	})
+	if tx.Error != nil {
+		return tx.Error
+	}
+
+	return nil
+}
+
+func (s sqlLite) SyncTeritoriKey(teritoriPubkey string, bertyPubkey string) error {
+	db := s.db
+
+	user := User{BertyPubKey: bertyPubkey}
+	tx := db.FirstOrCreate(&user, "berty_pub_key = ?", bertyPubkey)
+	if tx.Error != nil {
+		return tx.Error
+	}
+
+	tx = db.Create(&TeritoriKey{PubKey: teritoriPubkey, UserId: user.ID})
+	if tx.Error != nil {
+		return tx.Error
+	}
+
+	return nil
+}
+
+func (s sqlLite) AddChannel(workspaceName string, channelName string, bertyGroupLink string) error {
+	db := s.db
+
+	var workspace Workspace
+	tx := db.Where(Workspace{Name: workspaceName}).FirstOrCreate(&workspace)
+
+	var channel Channel
+	tx = db.Where(&Channel{
+		ChannelName: channelName,
+		BertyLink:   bertyGroupLink,
+		Wid:         workspace.ID,
+	}).FirstOrCreate(&channel)
+	if tx.Error != nil {
+		return tx.Error
+	}
+
+	return nil
+}
+
+func (s sqlLite) AddWorkspace(workspaceName string) error {
+	db := s.db
+	var workspace Workspace
+	tx := db.Where(Workspace{Name: workspaceName}).FirstOrCreate(&workspace)
+	if tx.Error != nil {
+		return tx.Error
+	}
+
+	return nil
+}
+
+func (s sqlLite) ListWorkspaces() ([]string, error) {
+	var workspaces []Workspace
+	_ = s.db.Find(&workspaces)
+
+	var workspaceIDs []string
+	for _, workspace := range workspaces {
+		workspaceIDs = append(workspaceIDs, workspace.Name)
+	}
+
+	return workspaceIDs, nil
+}
+
+func (s sqlLite) ListUsers() ([]User, error) {
+	var users []User
+	_ = s.db.Find(&users)
+
+	return users, nil
+}
+
+func (s sqlLite) ListChannels(workspaceName string) ([]string, error) {
+	var workspace Workspace
+	_ = s.db.Where("name = ?", workspaceName).First(&workspace)
```

Comment:
```suggestion
	_ = s.db.Where(Workspace{Name: workspaceName}).First(&workspace)
```

This use go types instead of string.

Your compiler can warn you if you break something.